[Andrew22Teoh] iP

Andrew22Teoh opened the pull request:
# Rizzler
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://en.wikipedia.org/wiki/Gullibility)) 

Rizzler frees you of having to keep track of your to-do list! It's:
- text-based,
- easy to learn, and
- ~~FAST~~ *SUPER FAST* to use!

All you have to do is
1. download it from [here](https://github.com/Andrew22Teoh/ip)
2. double-click it
3. add your tasks
4. and let it handle everything for you! 😉 😏 

The best part? It's **absolutely _FREE_**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (maybe eventually...)
- [ ] Will not flirt with you! (at some point?)
## 
Any burgeoning Java programmers can also use this to practice Java! Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Andrew22Teoh committed:
Replace Duke.java with Rizzler.java

Andrew22Teoh committed:
Rizzler.java: add in line separations

Andrew22Teoh committed:
Rizzler.java: update echo function

Andrew22Teoh committed:
Create Speech.java, RizzlerSpeech.java, InputLog.java
Delegate logic to other classes

Andrew22Teoh committed:
Create Task.java

Andrew22Teoh committed:
Restructure InputLog.java to TaskLog.java to work with Task.java

Andrew22Teoh committed:
Enable Level 3 functionality

Andrew22Teoh committed:
Delete unused UserSpeech.java

Andrew22Teoh committed:
Create ToDo.java

Andrew22Teoh committed:
Improve code readability and logic

Andrew22Teoh committed:
Create working Deadline.java

Andrew22Teoh committed:
Create working Event.java

Andrew22Teoh committed:
Configure automated testing

Andrew22Teoh committed:
Enable edge case catching

Andrew22Teoh committed:
Rectify a regression error

Andrew22Teoh committed:
Update testing cases

Andrew22Teoh committed:
Create delete operation

Andrew22Teoh committed:
Record possible future extensions in extensionFeatures.txt

Andrew22Teoh committed:
Update .gitignore file: remove file that no longer exists

Andrew22Teoh committed:
Add Javadoc comments into Deadline.java

Andrew22Teoh committed:
Add Javadoc comments to RizzlerSpeech.java

Andrew22Teoh committed:
Edit switch-case indentations in Rizzler.java

Andrew22Teoh committed:
Update Javadoc comments in Deadline.java

Andrew22Teoh committed:
Create Storage.java to handle saving taskLog

Andrew22Teoh committed:
Update .gitignore: do not track taskStorage

Andrew22Teoh committed:
Modify files to enable task storage functionality

Andrew22Teoh committed:
Merge branch-7 for Storage functionality

Andrew22Teoh committed:
Create DateTimeParser.java to interconvert between strings and datetime

Andrew22Teoh committed:
Update runtest.sh input to consider Storage

Andrew22Teoh committed:
Modify tasks to take in datetime objects instead of strings

Andrew22Teoh committed:
Merge branch 'branch-Level-8'
Modify Tasks to accept DateTime objects instead of strings

Andrew22Teoh committed:
Update test inputs in preparation for A-MoreOOP

Andrew22Teoh committed:
Create *Command classes extending abstract Command.java

Andrew22Teoh committed:
Create Parser to process user input and create commands

Andrew22Teoh committed:
Modify classes in line with More OOP

Andrew22Teoh committed:
Merge branch 'branch-A-MoreOOP'
Modify code to be more in line with OOP Principles

Andrew22Teoh committed:
Move classes into their respective packages

Andrew22Teoh committed:
Modify runtest.sh to work with new package directories

Andrew22Teoh committed:
Organise classes into packages and add in necessary imports

Andrew22Teoh committed:
Merge branch 'branch-A-Packages'
Organise files in a hierarchical manner and place them into packages

Andrew22Teoh committed:
Merge branch 'add-gradle-support'
Enable gradle usage in master

e1121208 commented on src/main/java/rizzler/Rizzler.java:
```diff
@@ -0,0 +1,35 @@
+import rizzler.ui.RizzlerSpeech;
+import rizzler.ui.parser.Parser;
+
+public class Rizzler {
```
That's a very interesting name!

blackpanther9229 replied:
> That's a very interesting name!

By the way, have you done the Pull Request. I can not find yours.

e1121208 commented on src/main/java/rizzler/Storage.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    private static void createFile(File file) {
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            // don't create it
+            System.out.println("Cannot create storage file");
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private Task tsvToTask(String tsv) throws DateTimeParseException {
+        String[] tsvFields = tsv.split("\t");
+        Task newTask;
+        switch (tsvFields[0]) {
+        case "ToDo":
+            boolean todoIsDone = Boolean.parseBoolean(tsvFields[1]);
+            String todoDesc = tsvFields[2];
+            newTask = new ToDo(todoDesc, todoIsDone);
+            break;
+        case "Deadline":
+            boolean deadlineIsDone = Boolean.parseBoolean(tsvFields[1]);
+            String deadlineDesc = tsvFields[2];
+            String deadlineTime = tsvFields[3];
+            newTask = new Deadline(deadlineDesc, deadlineTime, deadlineIsDone);
+            break;
+        case "Event":
+            boolean eventIsDone = Boolean.parseBoolean(tsvFields[1]);
+            String eventDesc = tsvFields[2];
+            String eventStart = tsvFields[3];
+            String eventEnd = tsvFields[4];
+            newTask = new Event(eventDesc, eventStart, eventEnd, eventIsDone);
+            break;
+        default:
+            newTask = new ToDo("nullTask");
+        }
+        return newTask;
+    }
+}
```
Would make your code clearer and understandable to the user if you added some header comments to your methods

e1121208 commented on src/main/java/rizzler/task/TaskLog.java:
```diff
@@ -0,0 +1,79 @@
+        this.numTasks = 0;
+    }
+
+    public TaskLog() {
+        this(new Task[100]);
+    }
```
Would it be more desirable to use a list instead of an array?

e1121208 commented on src/main/java/rizzler/ui/parser/Parser.java:
```diff
@@ -0,0 +1,188 @@
+                if (!eventInfo.contains("/from")) {
+                    outputCommand = new NullCommand(
+                            "remember to include a \"/from [start]\" for this event!");
+                    break;
+                } else if (!eventInfo.contains("/to")) {
+                    outputCommand = new NullCommand("remember to include a \"/to [end]\" for this event!");
+                    break;
+                }
+                String[] eventInfoBySplit = eventInfo.split("/from");
+                String eventDesc = eventInfoBySplit[0].trim();
+                if (eventDesc.isEmpty()) {
+                    outputCommand = new NullCommand("your event is missing a description honey.");
+                    break;
+                }
+                String eventDuration = eventInfoBySplit[1];
+                if (eventDuration.split("/to").length < 2) {
+                    outputCommand = new NullCommand("events require a description, a start, and an end dear");
+                    break;
+                }
+                String eventStart = eventDuration.split("/to")[0].trim();
+                String eventEnd = eventDuration.split("/to")[1].trim();
+                outputCommand = new EventCommand(eventDesc, eventStart, eventEnd);
+            } catch (DateTimeParseException e) {
+                outputCommand = new NullCommand(e.getMessage());
+            } catch (Exception e) {
+                outputCommand = new NullCommand("dreadfully sorry darlin', but there's an unexpected error");
+            }
+            break;
+        default:
+            outputCommand = new NullCommand("sincerest apologies darlin', i don't recognise that command.");
+            break;
+        }
+        return outputCommand;
+
+    }
+
+    public void close() {
+        scanner.close();
+    }
+}
```
Honestly, love how neat it looks

e1121208 reviewed (commented):
All in all, well done!

Andrew22Teoh committed:
Modify build.gradle for future JUnit testing

Andrew22Teoh committed:
Update all commands to return String[] instead of void

Andrew22Teoh committed:
Modify main classes to work with new commands

Andrew22Teoh committed:
Merge branch 'branch-Isolate-Command-Printing'
Remove onus of printing from commands

Andrew22Teoh committed:
Create test classes for DeadlineCommand and Parser

Andrew22Teoh committed:
Edit jar name in build.gradle

Andrew22Teoh committed:
Add JavaDoc comments to almost all classes

Andrew22Teoh committed:
Tweak code to comply with coding standard

Andrew22Teoh committed:
Add find functionality

Andrew22Teoh committed:
Merge branch 'branch-A-JavaDoc'
Add JavaDoc comments for almost all classes

Andrew22Teoh committed:
Merge branch 'branch-A-CodingStandard'
Tweak files to adhere to coding standard.

Andrew22Teoh committed:
Merge branch 'branch-Level-9'
Enable find functionality for users in master branch

Andrew22Teoh committed:
Change access modifier on Task.java getDesc()

Andrew22Teoh committed:
Modify build.gradle and add files for checkstyle

Andrew22Teoh committed:
Modify necessary files to enable checkstyle and clear all requirements

Andrew22Teoh committed:
Merge branch 'branch-EnableCheckstyle'
Modify build.gradle to enable checkstyle checks, edit files to satisfy requirements

blackpanther9229 commented on src/main/java/rizzler/Storage.java:
```diff
@@ -0,0 +1,118 @@
+
+/**
+ * Handles reading and writing between TaskLogs and files.
+ * Default directory is <code>taskStorage/taskLog.tsv</code>, relative to where <code>Rizzler</code> is run.
```
Very helpful

blackpanther9229 commented on src/main/java/rizzler/command/FindCommand.java:
```diff
@@ -0,0 +1,44 @@
+                output.append(numMatches);
+                output.append(". ");
+                output.append(task);
+                output.append("\n");
```
I would prefer 
` output.append(numMatches)
                .append(". ")
                .append(task)
                .append("\n");`

blackpanther9229 commented on src/main/java/rizzler/task/TaskLog.java:
```diff
@@ -0,0 +1,115 @@
+        }
+    }
+
+    private void doubleLogSize() {
+        int newLogSize = 2 * this.log.length;
+        Task[] newLog = new Task[newLogSize];
+        System.arraycopy(this.log, 0, newLog, 0, this.log.length);
+        this.log = newLog;
+    }
```
Still bettter for this to have comment

blackpanther9229 commented on src/main/java/rizzler/ui/parser/Parser.java:
```diff
@@ -0,0 +1,201 @@
+            } catch (Exception e) {
+                outputCommand = new NullCommand("dreadfully sorry darlin', but there's an unexpected error");
+            }
+            break;
```
These cases are are quite lenghty, so I would recommend you to divide to smaller part and write comment for each.

blackpanther9229 commented:
Overall most of your projects follow the coing standard. The layout, naming and statement are all corrects. You write comments for almost all of your non-trivial class/ method and make good use of JavaDoc tools like `<code> ... </code>`. Weel done

wang-h-z commented on src/main/java/Deadline.java:
Good work on changing the logic such that Tasks can use LocalDate objects instead of Strings. Looks good to me!

wang-h-z commented on src/main/java/TaskLog.java:
Looks like you are following the Java Coding Standard to a T! Great job!

wang-h-z commented on src/main/java/Rizzler.java:
```diff
@@ -23,6 +23,34 @@ public static void main(String[] args) {
+                    rizzlerSpeech.say();
+                    if (userInputArr.length == 1) {
+                        rizzlerSpeech.say("what's the task number you want to delete?");
+                        rizzlerSpeech.say();
```
I feel like you can abstract these repeated .say() calls to a Ui object which can handle this logic.

wang-h-z commented on src/main/java/rizzler/Rizzler.java:
Good work on picking out the nitty gritty things to improve on your coding standard!

wang-h-z reviewed (commented):
Overall, your code is cohesive and makes sense to me! Thank you for taking the extra mile to be consistent in your commit messages and complying with the Java Coding Standard. Keep up the good work!

Andrew22Teoh committed:
Add GUI files into resources folder

Andrew22Teoh committed:
Enable basic GUI functionality

Andrew22Teoh committed:
Modify Rizzler to work with GUI

Andrew22Teoh committed:
Merge branch 'branch-Level-10'
Complete GUI functionality with bye and list saving

Andrew22Teoh committed:
Enable text wrapping and infinite dialog box height

Andrew22Teoh committed:
Change font to arial size 13

Andrew22Teoh committed:
Enable storage of strings as deadline and event datetimes

Andrew22Teoh committed:
Modify some classes to better fit the GUI

MainWindow: remove logic of when to end the program for better SLAP

Rizzler: create new method shouldEnd() for MainWindow to end program

Command: change access modifier of setShouldEnd() to private so that
only other commands will be able to use this method

Parser: remove private scanner, as it is no longer required based
on the GUI input method

Andrew22Teoh committed:
Change access modifier of Command constructors

Andrew22Teoh committed:
Delete Speech.java and RizzlerSpeech.java

Speech and RizzlerSpeech as output classes are no longer needed, since
the output strings are handled by the JavaFX ui. Formatting of output
strings is also no longer required, since visual identification of
which lines are from Rizzler and from the user are clear now.

Andrew22Teoh committed:
Add JavaDoc comments for DateTimeParser's canParse method

Andrew22Teoh committed:
Add HelpCommand for users to view available commands

Andrew22Teoh committed:
Delete unused ByeCommand

Andrew22Teoh committed:
Update JavaDoc comment for Parser class

Andrew22Teoh committed:
Fix dialogbox not going infinitely long

Andrew22Teoh committed:
Modify gitignore to not track extensionFeatures.txt

Andrew22Teoh committed:
Merge branch 'branch-AdaptToGui'

Delete unnecessary classes leftover from the text interface, and modify
other classes to clear redundant elements like scanner and printing.

Andrew22Teoh committed:
Fix checkstyle conflicts

Andrew22Teoh committed:
Enable assertion errors in build.gradle

Andrew22Teoh committed:
Add assertions to multiple classes

Andrew22Teoh committed:
Improve code quality for multiple classes

DeadlineCommand, EventCommand, TodoCommand: Extract the logic of creating
confirmation strings when a new task is generated into a separate method
to apply better SLAP.

Deadline, Task: modify javadoc comments

Andrew22Teoh committed:
Merge pull request #2 from Andrew22Teoh/branch-A-Assertions

Merge branch-A-Assertions

Andrew22Teoh committed:
Merge branch 'master' into branch-A-CodeQuality

Andrew22Teoh committed:
Merge pull request #4 from Andrew22Teoh/branch-A-CodeQuality

Merge branch-A-CodeQuality into master

Andrew22Teoh committed:
Merge pull request #5 from Andrew22Teoh/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

Andrew22Teoh committed:
Modify HelpCommand and parser to allow easier editing of output help messages

Andrew22Teoh committed:
Add HelpCommand to recognise user request for help

Andrew22Teoh committed:
Merge pull request #6 from Andrew22Teoh/branch-BCD-Extension

Merge BCD-Extension

Andrew22Teoh committed:
Add some assertions and minor edits to GUI files

Andrew22Teoh committed:
Split parseInput method up into smaller methods

Andrew22Teoh committed:
Fix syntax error with assertions

Andrew22Teoh committed:
Modify attribute order for checkstyle

Andrew22Teoh committed:
Merge branch 'branch-ImproveSlap'

Andrew22Teoh committed:
Update HelpCommand, remove dead code

Andrew22Teoh committed:
Create checkstyle reminder on Parser

Andrew22Teoh committed:
Modify Todo, Deadline, and Events

Improve JavaDoc comments for these methods, and allow Events to
take in dates that aren't properly formatted.

Andrew22Teoh committed:
Add JavaDoc comments to GUI elements

Andrew22Teoh committed:
Update JavaDoc comments for all commands

Andrew22Teoh committed:
Update Javadoc comments for tasks

Andrew22Teoh committed:
Update JavaDoc comments for parsers

Andrew22Teoh committed:
Remove unnecessary line break

Andrew22Teoh committed:
Abstract datetime conversion for better OOP

Andrew22Teoh committed:
Move magic number to static final attribute

Andrew22Teoh committed:
Fix minor styling issues in Rizzler

Andrew22Teoh committed:
Remove magic literals from Tasks

Andrew22Teoh committed:
Fix checkstyle error in Main

Andrew22Teoh committed:
Improve GUI experience

Andrew22Teoh committed:
Enhance SLAP in Parser and improve error message handling

Create static error messages for each error message, avoiding
magic literals in constructing error NullCommand.

Extract methods from the parseInput() method, reducing
method length significantly and improving readability.

Andrew22Teoh committed:
Change access of getType() to protected

Andrew22Teoh committed:
Remove magic literals in Parser

Andrew22Teoh committed:
Modify FindCommand to use java streams

Andrew22Teoh committed:
Merge branch 'branch-A-Streams'

Andrew22Teoh committed:
Merge branch 'master' into branch-AdjustForGrading

Andrew22Teoh committed:
Add multiple search term functionality  into FindCommand

Andrew22Teoh committed:
Fix checkstye errors on FindCommand

Andrew22Teoh committed:
Create static attribute for 1-based indexing

Andrew22Teoh committed:
Delete unused GreetCommand

Andrew22Teoh committed:
Edit HelpCommand to be slightly more descriptive

Andrew22Teoh committed:
Add break lines between JavaDoc comment and params

Andrew22Teoh committed:
Remove unnecessary assertion from Storage

Andrew22Teoh committed:
Add inverted commas for unrecognised commands

Andrew22Teoh committed:
Merge branch 'branch-AdjustForGrading'

Improve JavaDoc comments to adhere to coding standard.
Improve SLAP by extracting many methods and minimising method length.
Improve GUI and command handling by handling whitespace better,
and beautifying GUI

Andrew22Teoh committed:
Move packages around for better structure

Andrew22Teoh committed:
Update package imports appropriately

Andrew22Teoh committed:
Add final to an attribute in Task

Andrew22Teoh committed:
Update and add JUnit tests for more classes

Andrew22Teoh committed:
Fix minor checkstyle error

Andrew22Teoh committed:
Add bot name to GUI window title

Andrew22Teoh committed:
Add Ui.png to docs folder

Andrew22Teoh committed:
Update README.md

Andrew22Teoh committed:
Update README.md

Andrew22Teoh committed:
Update README.md

Andrew22Teoh committed:
Update README.md
